great_expectations/rule_based_profiler/helpers/util.py

```diff
@@ -156,10 +156,17 @@ def get_batch_ids(
 
     batch_ids: List[str] = [batch.id for batch in batch_list]
 
+    num_batch_ids: int = len(batch_ids)
+
     if limit is not None:
-        batch_ids = batch_ids[0:limit]
+        if not (isinstance(limit, int) and (0 <= limit <= num_batch_ids)):
```

Comment:
Do we need the `isinstance` check? The type is declared to be an `Optional[int]`. Alternatively, we could remove the `None` check and just have this.

Reply:
@billdirks We can remove, thanks to static type checking -- I just wanted extra robustness, but will remove (with comment).